src/utils.cpp

```diff
@@ -109,3 +130,163 @@ KeyRingReceive ZypperUtils::keyReport()
     return s_keyReceiveReport;
 }
 
+/************************************* ZYPPER INSTALL ************************************/
+void ZypperUtils::markPackagesForInstallation( const string& repoUrl, const string& packageName )
+{
+    Pathname sysRoot( "/" );
+    // acquire initial zypp lock
+    s_zypp = getZYpp(); 
+    
+    makeNecessaryChangesToRunAsRoot( sysRoot );
+    initTarget( sysRoot );
+    initRepoManager( repoUrl, packageName );
+    
+    //TRUST_AND_IMPORT_KEY - we are adding this to the system repos
+    KeyRing::setDefaultAccept( KeyRing::TRUST_AND_IMPORT_KEY );
+    cout << "TRUST_AND_IMPORT_KEY accepted" << endl;
+    refreshRepoManager();
+    
+    //load cache
+    try {
+	cout << "Loading resolvables from " << s_repoInfo.alias() << endl;
+	s_repoManager->loadFromCache( s_repoInfo );      
+    }
+    catch (const Exception& exp) {
+	// cachefile has old format or is corrupted: try to rebuild it
+	s_repoManager->cleanCache( s_repoInfo );
+	s_repoManager->buildCache( s_repoInfo );
+	s_repoManager->loadFromCache( s_repoInfo );
+    }
+    
+    // select/specify package(s) to install
+    PoolQuery q;
+    q.addKind( ResKind::package );
+    q.addAttribute( sat::SolvAttr::name, packageName );
+    q.setMatchExact();
+    
+    PoolItemBest bestMatches(q.begin(), q.end());    
+    if ( !bestMatches.empty() ) {
+	for ( PoolItemBest::iterator it = bestMatches.begin(); it != bestMatches.end(); ++it ) {
+	    ui::asSelectable()( *it )->setOnSystem( *it, ResStatus::USER );
+	}
+    }
```

Comment:
Thank you so much for providing important insights. I will these into consideration and make changes to the code accordingly.